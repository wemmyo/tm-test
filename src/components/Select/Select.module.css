/* .select {
  padding: 0.67857143em 1em;
  background: #fff;
  border: 1px solid rgba(34, 36, 38, 0.15);
  color: rgba(0, 0, 0, 0.87);
  border-radius: 0.28rem;
} */
.inputGroup {
  margin-bottom: 1rem;
}
.label {
  display: inline-block;
  margin-bottom: 0.5rem;
}
.select {
  /* display: block; */
  width: 100%;
  padding: 0.6em 1em;
  background: #fff;
  border: 1px solid rgba(34, 36, 38, 0.15);
  color: rgba(0, 0, 0, 0.87);
  border-radius: 0.28rem;
  transition: box-shadow 0.1s ease, border-color 0.1s ease;
  box-sizing: border-box;
}
