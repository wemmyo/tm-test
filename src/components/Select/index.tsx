import * as React from "react";
import styles from "./Select.module.css";

interface SelectProps extends React.SelectHTMLAttributes<HTMLSelectElement> {
  label?: string;
}

const Select: React.FC<SelectProps> = ({ children, label, ...props }) => {
  return (
    <div className={styles.inputGroup}>
      {label && <label className={styles.label}>{label}</label>}
      <select {...props} className={styles.select}>
        {children}
      </select>
    </div>
  );
};

export default Select;
