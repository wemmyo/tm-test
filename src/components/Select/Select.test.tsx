import React from "react";
import { shallow } from "enzyme";
import Select from "./index";

it("renders without crashing", () => {
  expect(shallow(<Select label="Test label" />)).toMatchSnapshot();
});
