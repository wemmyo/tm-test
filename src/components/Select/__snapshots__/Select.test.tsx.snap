// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="inputGroup"
>
  <label
    className="label"
  >
    Test label
  </label>
  <select
    className="select"
  />
</div>
`;
